Id b 20

EugenTovarchi opened the pull request:
Даже с правильной конфигурацией проблема с .Include в репозиториях методов GetById (хотя мб там оно и не надо).

Посмотри пожалуйста на пустые конструкторы в Breed, Species. Нужны ли они там если допустим в Breed мы добавляем 
[JsonConstructor] ? 

Без dataSourceBuilder.EnableDynamicJson(); в ReadDbContext ошибки вылетают при Get запросах. 

И так ли важно переделывать из VO в классы ?

EugenTovarchi committed:
настроил конфигурации для species  / pet
сделал новые св-ва в Pet для Species / Breed
сделал ReadSpeciesRepository на будущее

EugenTovarchi committed:
решил проблему с конфигурацией и репо

SachkovK commented:
Vo можно не делать классами, можно для VO просто оставить record

SachkovK commented:
не знаю точно где ошибка в конфигурациях, надо покатаься потому что запросы должны работать без костылей

EugenTovarchi committed:
нашёл альтернативу dataSourceBuilder.EnableDynamicJson();
это вариант из видео Кирилла - ручная конфигурация десериализации(забыл про неё)

EugenTovarchi commented:
Фикс для обнаруженной проблемы был сделан в параллельной ветке 21.  Смержил изменения в эту ветку - теперь PR содержит все исправления. 

Тут получается 2й вариант нашёл т.к. забыл внести ручные настройки как у тебя в SpeciesDtoConfiguration
var dataSourceBuilder = new NpgsqlDataSourceBuilder(connectionString);
// альтернатива настройке конфига в SpeciesDtoConfiguration для breeds
//dataSourceBuilder.EnableDynamicJson(); 

в SpeciesDtoConfiguration просто добавлено свойство при этом:
 builder.Property(s => s.Breeds)
     .HasColumnType("jsonb")
     .HasColumnName("breeds");

твоя альтернатива 
 builder.Property(s => s.Breeds)
    .HasConversion(
        breeds => JsonSerializer.Serialize(string.Empty, JsonSerializerOptions.Default),
        json => JsonSerializer.Deserialize<BreedDto[]>(json, JsonSerializerOptions.Default)!
    )
    .HasColumnType("jsonb")
    .HasColumnName("breeds");
    
    Работает и так и так.

SachkovK commented:
отлично
